ecosystem/sep-0012.md

```diff
@@ -102,6 +102,7 @@ Name | Type | Description
 `memo` | string | (optional) the client-generated [memo](https://developers.stellar.org/docs/glossary/transactions/#memo) that uniquely identifies the customer. If a memo is present in the decoded SEP-10 JWT's `sub` value, it must match this parameter value. If a muxed account is used as the JWT's `sub` value, memos sent in requests must match the 64-bit integer subaccount ID of the muxed account. See the [Shared Accounts](#shared-omnibus-or-pooled-accounts) section for more information.
 `memo_type` | string | (**deprecated**, optional) type of `memo`. One of `text`, `id` or `hash`. Deprecated because memos should always be of type `id`, although anchors should continue to support this parameter for outdated clients. If `hash`, `memo` should be base64-encoded. If a memo is present in the decoded SEP-10 JWT's `sub` value, this parameter can be ignored. See the [Shared Accounts](#shared-omnibus-or-pooled-accounts) section for more information.
 `type` | string | (optional) the type of action the customer is being KYCd for.  See the [Type Specification](#type-specification) below.
```

Comment:
> it doesn't make sense to use both in the same request.

I think it still does. If we keep type to the protocol level (sep-24, sep-6, sep-31) it makes either for the anchor to work with the request.
First, it's technically possible for transaction to have the same id between protocols
Second, in AP implementation, without specifying type, it'll need to do 3 requests, instead of 1 (because it needs to search for transaction in all 3 tables).

So, IMO we should instead advice into matching `type` with the protocol that's being used, when defining transaction, and deprecate all other usages.
WDYT?

Reply:
I don't think anyone will use `type` moving forward if we add `transaction_id`. For those that already using `type`, limiting its usage to indicate which SEP the customer is KYC'ing for would require them to update their implementation, and at that point they're probably better off switching to `transaction_id`.

As for the uniqueness of the ID, I think that is up to the anchor to ensure the IDs aren't the same between the SEPs they support. The anchor platform chose to use different tables for each SEP (which I don't think is the right approach), but they're random UUID4's so chance of collision is minimal.

Reply:
`type` is still required for SEP-31; and we can use `type` to specify protocol when used with `transaction_id`

Reply:
Gleb and I discussed this offline. In short: we'll not deprecate `type` since the SEP-31 has not been updated to create transactions before KYC is checked / processed. 

However, I do think we should deprecated `type` once SEP-31 has been updated, since there is no need for general-purpose custom field, and we can add an alternative to `type` that is explicitly for the purpose of clarifying the SEP, like a `sep` or `transaction_type` argument.